### DISCUSSION TITLE
warp GET benchmarking on s3 api fails

### DISCUSSION THREAD

**Discussion by dpower5** [seaweedfs/seaweedfs#2134] [General]
Im unable to run the warp GET benchmarks on the s3 api endpoint for the seaweedfs, while I have no issue with the PUT tests

command:
./warp get --duration=30s --obj.size=100k -objects=200 --host=endpoint --access-key=acc-key --secret-key=sec-key-xxx

output
warp: <ERROR> unexpected download size. want:100000, got:100265
warp: <ERROR> unexpected download size. want:100000, got:100265
warp: <ERROR> unexpected download size. want:100000, got:100265
warp: <ERROR> unexpected download size. want:100000, got:100265
.
.
.
.
warp: <ERROR> unexpected download size. want:100000, got:100265

----------------------------------------
Operation: PUT
* Average: 0.63 MiB/s, 6.56 obj/s

Throughput, split into 28 x 1s:
 * Fastest: 791.7KiB/s, 8.11 obj/s
 * 50% Median: 636.8KiB/s, 6.52 obj/s
 * Slowest: 558.8KiB/s, 5.72 obj/s

----------------------------------------
Operation: GET
Errors: 1231

----------------------------------------

thus no benchmarks are generated for GET dues to mismatching file size.
seems like somehow the 100000 bytes(100 kb) file  on the seaweed is changing to 100265 bytes.
I wanted to know if anyone else faced this before creating an issue.

the same file when downloaded back to disk using s3 api is 100kb
This is very important as I want to read my ML data from seaweedfs(using s3 api, without downloading to disk) thru tensor flow dataset api.

```
train_dataset = tf.data.TFRecordDataset(filenames=[
    "s3://bucketname/path/to/file1.tfrecord",
    "s3://bucketname/path/to/file2.tfrecord",
]).map(record_parser).batch(BATCH_SIZE)

```
 it errors out complaining about corrupt data (The same logic works fine on an AWS s3 bucket), and also the data appears to be fine once downloaded to local disk and the ML pipeline works fine once the data is on local disk , like below snippet.

```
// "local/path/" contains the same records downloaded from seaweedfs thru s3 api
train_dataset = tf.data.TFRecordDataset(filenames=[
    "local/path/file1.tfrecord",
    "local/path/file2.tfrecord",
]).map(record_parser).batch(BATCH_SIZE)
```

**Comment by chrislusf** [comment]
Make sure the `weed s3 -config` or `s3.configure` in `weed shell` is used.

**Reply by dpower5** [reply]
the s3 has been configured properly and we have a fully functional working s3 api and we have verified it. Its just that the behavior of differing file size on the storage and thus failing warp GET benchmarks. Is there any other setting that should be looked into .

**Reply by chrislusf** [reply]
No. Usually it is the access and secret key.

```
$ weed shell
master: localhost:9333 filer: localhost:8888
> s3.configure
{
  "identities": [
    {
      "name": "admin",
      "credentials": [
        {
          "accessKey": "any",
          "secretKey": "any"
        }
      ],
      "actions": [
        "Admin"
      ]
    }
  ]
}
```

```
$ warp get --duration=30s --obj.size=100k -objects=200 --host=localhost:8333 --access-key=any --secret-key=any
warp: Benchmark data written to "warp-get-2021-06-17[175104]-SOLg.csv.zst"

Skipping PUT too few samples.

-------------------

Operation: GET
* Average: 140.18 MiB/s, 1469.90 obj/s

Throughput, split into 29 x 1s:
 * Fastest: 158.8MiB/s, 1665.35 obj/s
 * 50% Median: 142.4MiB/s, 1492.81 obj/s
 * Slowest: 108.8MiB/s, 1140.67 obj/s
warp: Cleanup Done
```

**Reply by dpower5** [reply]
thank you. that solved it

**Reply by dpower5** [reply]
Out of curiosity, what metadata was being added to the files(in the default env without s3 configured access and secret keys).
